checks.d/consul.py

```diff
@@ -0,0 +1,215 @@
+                                              self.init_config.get('catalog_checks'))
+
+        try:
+            health_state = self.consul_request(instance, '/v1/health/state/any')
+
+            STATUS_SC = {
+                'passing': AgentCheck.OK,
+                'warning': AgentCheck.WARNING,
+                'critical': AgentCheck.CRITICAL,
+            }
+
+            for check in health_state:
+                status = STATUS_SC.get(check['Status'])
+                if status is None:
+                    continue
+
+                tags = ["check:{0}".format(check["CheckID"])]
+                if check["ServiceName"]:
+                    tags.append("service:{0}".format(check["ServiceName"]))
+                if check["ServiceID"]:
+                    tags.append("service-id:{0}".format(check["ServiceID"]))
+
+            services = self.get_services_in_cluster(instance)
+            self.service_check(self.HEALTH_CHECK, status, tags=tags)
+
+        except Exception as e:
+            self.service_check(self.CONSUL_CHECK, AgentCheck.CRITICAL,
+                               tags=service_check_tags)
+        else:
+            self.service_check(self.CONSUL_CHECK, AgentCheck.OK,
+                               tags=service_check_tags)
+
+        if perform_catalog_checks:
+            services = self.get_services_in_cluster(instance)
+            if service_whitelist:
+                services = [s for s in services if s in service_whitelist]
+            else:
+                #Default to polling all services
+                self.log.warn('Consul service whitelist not defined. Agent will poll for all %d services found' % len(services))
+
+            nodes = self.get_nodes_in_cluster(instance)
+
+            main_tags = []
+            agent_dc = self._get_agent_datacenter(instance)
+            if agent_dc is not None:
+                main_tags.append('consul_datacenter:{0}'.format(agent_dc))
+
+            nodes_to_services = {}
+
+            self.gauge('consul.catalog.services_up', len(services), tags=main_tags)
+            self.gauge('consul.catalog.nodes_up', len(nodes), tags=main_tags)
+
+            for service in services:
+                nodes_with_service = self.get_nodes_with_service(instance, service)
+                node_tags = ['consul_service_id:{0}'.format(service)]
+
+                self.gauge('consul.catalog.nodes_up',
+                           len(nodes_with_service),
+                           tags=node_tags)
+
+                for n in nodes_with_service:
+                    node_id = n.get('Node') or None
+
+                    if not node_id:
+                        continue
+
+                    if node_id in nodes_to_services:
+                        nodes_to_services[node_id].append(service)
+                    else:
+                        nodes_to_services[node_id] = [service]
+
+            for node, services in nodes_to_services.items():
```

Comment:
With python2, you should use .iteritems() instead of .items() to avoid doing a copy of the dictionary.